src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrayOfAcrions = [];
+  let currentAction = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        let addAction = { ...currentAction };
+
+        addAction = Object.assign(addAction, action.extraData);
+
+        arrayOfAcrions.push(addAction);
+        currentAction = addAction;
+        break;
+
+      case 'removeProperties':
+        const removeAction = { ...currentAction };
+
+        for (const property of action.keysToRemove) {
+          delete removeAction[property];
+        }
+        arrayOfAcrions.push(removeAction);
+        currentAction = removeAction;
+        break;
+
+      default:
+        arrayOfAcrions.push({});
+        currentAction = {};
```

Comment:
There's no `break` statement in the default case.

I think you should create separate case for `clear` action and leave default empty
```suggestion
      case 'clear':
        currentAction = {};
        break;

      default:
        break;
```